src/core/evaluator.js

```diff
@@ -2264,36 +2260,62 @@ class PartialEvaluator {
       }
 
       const currentTransform = getCurrentTextTransform();
-      const posX = currentTransform[4];
-      const posY = currentTransform[5];
-      const lastPosX = textContentItem.prevTransform[4];
-      const lastPosY = textContentItem.prevTransform[5];
+      let posX = currentTransform[4];
+      let posY = currentTransform[5];
+      let lastPosX = textContentItem.prevTransform[4];
+      let lastPosY = textContentItem.prevTransform[5];
 
       if (lastPosX === posX && lastPosY === posY) {
         return;
       }
 
-      const advanceX = (posX - lastPosX) / textContentItem.textAdvanceScale;
-      const advanceY = (posY - lastPosY) / textContentItem.textAdvanceScale;
+      if (rotate !== 0) {
```

Comment:
I'm not entirely sure that I understand why the *page* rotation really matters here, since a PDF document can easily specify the current transformation matrix and/or the text matrix such all text is rendered at an (arbitrary) angle to the page itself?

Reply:
I added that to handle `issue6019.pdf`: visually we've LTR strings but if we omit the rotation it's TTB (or BTT, I don't remember) strings.
So here I finally just "apply" the rotation matrix to the current text matrix.
It was working ok before my patch because we just got chars from a stream without really thinking about their positions.
Anyway, from a user pov, there are LTR strings.